tools/github_webhook_processor.php

```diff
@@ -270,7 +283,108 @@ function handle_pr($payload) {
 		
 	$msg = '['.$payload['pull_request']['base']['repo']['full_name'].'] Pull Request '.$action.' by '.htmlSpecialChars($payload['sender']['login']).': <a href="'.$payload['pull_request']['html_url'].'">'.htmlSpecialChars('#'.$payload['pull_request']['number'].' '.$payload['pull_request']['user']['login'].' - '.$payload['pull_request']['title']).'</a>';
 	sendtoallservers('?announce='.urlencode($msg), $payload);
+}
 
+//creates a comment on the payload issue
+function create_comment($payload, $comment){
+	global $apiKey;
+	$scontext = array('http' => array(
+		'method'	=> 'POST',
+		'header'	=>
+			"Content-type: application/json\r\n".
+			'Authorization: token ' . $apiKey,
+		'ignore_errors' => true,
+		'user_agent' 	=> 'tgstation13.org-Github-Automation-Tools',
+		'content' => json_encode(array('body' => $comment))
+	));
+	echo file_get_contents($payload['pull_request']['comments_url'], false, stream_context_create($scontext));
+}
+
+//returns the payload issue's labels as a flat array
+function get_pr_labels_array($payload){
+	$issue = json_decode(apisend($url));
+	$result = array();
+	if(isset($issue['labels']))
+		foreach($issue['labels'] as $l)
+			$result[] = $l['name'];
+	return $result;
+}
+
+//helper for getting the path the the balance json file
+function pr_balance_json_path(){
+	return $prBalanceJson != '' ? $prBalanceJson : 'pr_balances.json';
+}
+
+//return the assoc array of login -> balance for prs
+function pr_balances(){
+	$path = pr_balance_json_path();
+	if(file_exists($path))
+		return json_decode(file_get_contents($path), true);
+	else
+		return array();
+}
+
+//returns 1 if it's a fix/feature/etc, 0 if it's a neutral, -1 if it's a Feature/Tweak/Balance
+function get_pr_code_friendliness($payload, $oldbalance){
+	global $startingPRBalance;
+	$labels = get_pr_labels_array($payload);
+	//anything not in this list defaults to -1
```

Comment:
should default to doing nothing, and have both positive (already done) and negative need to be explicitly set